OpenOversight/app/api/v1.py

```diff
@@ -0,0 +1,147 @@
+
+    if assignments is None:
+        assignments = (
+            db.session.query(Assignment)
+            .join(Assignment.base_officer)
+            .filter(Officer.department_id == department_id)
+            .options(contains_eager(Assignment.base_officer))
+            .options(joinedload(Assignment.unit))
+            .options(joinedload(Assignment.job))
+            .all()
+        )
+        put_database_cache_entry(*cache_params, assignments)
+
+    return objs_to_dicts_jsonify(assignments)
+
+
+@v1.route("/departments/<int:department_id>/incidents", methods=[HTTPMethod.GET])
+@limiter.limit("5/minute")
+def get_dept_incidents(department_id: int) -> Response:
+    cache_params = (Department(id=department_id), KEY_DEPT_ALL_INCIDENTS)
+    incidents = get_database_cache_entry(*cache_params)
+
+    if incidents is None:
+        incidents = Incident.query.filter_by(department_id=department_id).all()
+        put_database_cache_entry(*cache_params, incidents)
+
+    return objs_to_dicts_jsonify(incidents)
+
+
+@v1.route("/departments/<int:department_id>/salaries", methods=[HTTPMethod.GET])
+@limiter.limit("5/minute")
+def get_dept_salaries(department_id: int) -> Response:
+    cache_params = (Department(id=department_id), KEY_DEPT_ALL_SALARIES)
+    salaries = get_database_cache_entry(*cache_params)
+
+    if salaries is None:
+        salaries = (
+            db.session.query(Salary)
+            .join(Salary.officer)
+            .filter(Officer.department_id == department_id)
+            .options(contains_eager(Salary.officer))
+            .all()
+        )
+        put_database_cache_entry(*cache_params, salaries)
+
+    return objs_to_dicts_jsonify(salaries)
+
+
+@v1.route("/departments/<int:department_id>/links", methods=[HTTPMethod.GET])
+@limiter.limit("5/minute")
+def get_dept_links(department_id: int) -> Response:
+    cache_params = (Department(id=department_id), KEY_DEPT_ALL_LINKS)
+    links = get_database_cache_entry(*cache_params)
+
+    if links is None:
+        links = (
+            db.session.query(Link)
+            .join(Link.officers)
+            .filter(Officer.department_id == department_id)
+            .options(contains_eager(Link.officers))
+            .all()
+        )
+        put_database_cache_entry(*cache_params, links)
+
+    return objs_to_dicts_jsonify(links)
+
+
+@v1.route("/departments/<int:department_id>/descriptions", methods=[HTTPMethod.GET])
+@limiter.limit("5/minute")
+def get_dept_descriptions(department_id: int) -> Response:
+    cache_params = (Department(id=department_id), KEY_DEPT_ALL_NOTES)
+    notes = get_database_cache_entry(*cache_params)
+
+    if notes is None:
+        notes = (
+            db.session.query(Description)
+            .join(Description.officer)
+            .filter(Officer.department_id == department_id)
+            .options(contains_eager(Description.officer))
+            .all()
+        )
+        put_database_cache_entry(*cache_params, notes)
```

Comment:
It might be worth moving these to static functions on each of the objects since they are used both here and in views.